infor-design/enterprise
# 3111 - Fix theme switching and css layouts and date display bugs [v4.23.x]

**Explain the _details_ for making this change. What existing problem does the pull request solve?**

Fixes several more QA issues as noted in the tests. This PR can now be reviewed but the tests will fail until previous PR's are merged.

**Related github/jira issue (required)**:
Fixes #3111 

**Steps necessary to review your pull request (required)**:

**Testing 3111*
- go to http://localhost:4000/components/monthview/example-index.html and http://localhost:4000/components/calendar/example-index.html
- switch them back and forth between vibrant and subtle
- should look ok for layout

**Testing 3110*
- go to http://localhost:4000/components/monthview/example-index.html and http://localhost:4000/components/calendar/example-index.html
- focus the month picker in the header
- try typing (should not be able to now as its a span)

**Testing 3107**
- go to http://localhost:4000/components/calendar/example-index.html
- select any day like : November 13, 2019
- click + icon at the top right, then click "Add Event(Modal)"
- In the Request Time Off modal, uncheck All Day checkbox
- Select dates: from 11/11/2019 to 11/13/2019
- Select time: from 12:00 AM to 12:00 PM
- Click Submit button
- See that the date is now displayed in decimals
- Also added the time to the display on the right for some situations, click around on the events. If the event is just hours, then hours will be shown in the range. If the date is decimals/partial day then hours will be shown in the range. If the event is all day long no hours are shown.
